### DISCUSSION TITLE
AqualinkD Controlled Color Lights Not Working

### DISCUSSION THREAD

**Discussion by superdupe** [aqualinkd/AqualinkD#195] [General]
I can't get AqualinkD to properly control my lights, but I'm getting close...  With one significant problem... 

I **think** I have figured out how this is supposed to work.  For AqualinkD to actually send the pulses and control the lights, we first have to make sure the lights are not set as color in the control panel.  **Check**

Next, we use the lightMode=0 to tell AqualinkD that it is controlling the light... My config file light entries are as follows.  **Check**

button_03_label=Pool Light North
button_03_lightMode=0

button_04_label=Pool Light South
button_04_lightMode=0

button_05_label=Spa Light
button_05_lightMode=0

Of course, we also have to set the timing of the pulses in the config file.  I'm not sure why we mess with the timing unless it has something to do with my issue... But regardless, this is what I'm currently using...  **Check**

light_programming_mode=.90

light_programming_initial_on=15

light_programming_initial_off=12

Finally, we have to make entries in the web page config.js file so when we long press on the light, a menu appears.  There was very little info on this so I **think** I have it right.  Please someone let me know if this is not correct.

In the config.js file, I've listed the following.  My guess is the order on the list corresponds with the amount of pulses the AqualinkD system will send.  So my list is exactly how my lights cycle, taken directly from the programming instructions.  This should be the case of any colored light, and follow the manufacturers instructions.  Mine are J&J ColorSplash XG which are setup just like the Pentair Intellibrite.

 // This get's picked up by dynamic_config.js and used as mode 0
      var light_program = [
        "Peruvian Paradise - Show",
        "Super Nova - Show",
        "Northern Lights - Show",
        "Tidal Wave - Show",
        "Patriot Dream - Show",
        "Desert Skies - Show",
        "Nova - Show",
        "Parisian Blue",
        "NZ Green",
        "Brazilian Red",
        "Arctic White",
        "Miami Pink",
        "Color Lock",
        "Return"
      ];

So here is my problem, for some reason, I have to pick the color or show below the one I actually want.  For example, I want Arctic White.  If I select Arctic White, the system sends 10 pulses and I get Brazilian Red.  However, if I pick Miami Pink, it will indeed pulse 11 times and land on Arctic White.  So either I'm way off on how this is supposed to work, or there is something in the code or my pulse timing that is way off.  I've tested it a few times.  Another example, if I pick New Zealand Green, I get Parisian Blue.  But if I pick Brazilian Red, I get New Zealand Green.  Its like the system is always one pulse behind.

If I manually go to the control panel and flip the relay on and off myself, 11 times, I get Arctic White or whatever color I'm trying to achieve.  So I'm 99% sure the list and order are correct.  The question is why AqualinkD is not sending the right amount of pulses... 

Ok, phew... This is where I'm at... Looking for help or direction with where to go next...  Could it be timing or is there something else in the code I'm not getting?

Thanks

**Comment by sfeakes** [comment]
Everything you are doing and the configuration sounds correct.  It’s probably just how your specific light works.
Lights that I’ve tested / have pre-configured will go to the 1st program once you get it into the programming mode.  IE, once it’s in programming mode for you light it would be displaying  “Peruvian Paradise”, so it would only need 10 cycles to get to “Arctic white”.  Sounds like your specific light starts programming mode in some other state, so it needs 1 cycle to get to “ Peruvian Paradise” and 11 to get to “Arctic White”.   To overcome this I would simply add “Last Color” (or something) to the beginning of the list of colors in config.js.   See if that works.

**Reply by superdupe** [reply]
Thank you for the reply.  :)  I'll give that a try.

These are J&J Electronics ColorSplash XG-W Series RGB + White LED lights.

According to their own literature, they are programmed exactly like the Pentair Intellibrite 5G lights.  Some evidence I have found on the internet suggests they are owned by the same company... Anyway, the literature says they are programmed the same with the exception the names of the colors are different.  In fact it also talks about how to configure them as Pentair lights in the Aqualink controller.  

Have you or anyone else used the Pentair Intellibrite lights?  Curious to see if my problem is consistent... And thanks for this program!  I'm using it to control the lights through automation at different times, something the Jandy controller won't let me do...

**Reply by superdupe** [reply]
Ok, this appears to have worked!  I added a random name on top of my color list, and it now seems to be sending the correct amount of pulses.  For anyone else with these lights, here is the entry in my config.js:

// This get's picked up by dynamic_config.js and used as mode 0
      var light_program = [
	"Programming Mode",        
	"Peruvian Paradise - Show",
        "Super Nova - Show",
        "Northern Lights - Show",
        "Tidal Wave - Show",
        "Patriot Dream - Show",
        "Desert Skies - Show",
        "Nova - Show",
        "Parisian Blue",
        "NZ Green",
        "Brazilian Red",
        "Arctic White",
        "Miami Pink",
        "Color Lock",
        "Return"
      ];
